### ISSUE TITLE
A full non-relative URL in module-shims?

### ISSUE THREAD

**Issue by Hypercubed** [guybedford/es-module-shims#6]
Is there a technical reason that full non-relative URLs don't work inside shimmed modules?

For example:

```
<script defer src="https://unpkg.com/es-module-shims@0.2.0/dist/es-module-shims.js"></script>
<script type="importmap-shim">
{
  "imports": {
    "lit-html": "https://unpkg.com/lit-html@1.0.0/lit-html.js"
  }
}
</script>
<script type="module-shim">
  import { html } from "lit-html";                                             // works
  // import { html } from "https://unpkg.com/lit-html@1.0.0/lit-html.js";      // does not work
  console.log(html);
</script>
```

Codepen for testing: https://codepen.io/Hypercubed/pen/daLrMZ

The error is:
```
Uncaught (in promise) Error: Unable to resolve bare specifier "https://unpkg.com/lit-html@1.0.0/lit-html.js" from https://s.codepen.io/boomerang/iFrameKey-d5cb004a-1e37-cbae-c83b-d4327172eec9/
    at throwBare (es-module-shims.js:174)
    at resolveImportMap (es-module-shims.js:170)
    at resolve (es-module-shims.js:925)
    at load.d.Promise.all.deps.map (es-module-shims.js:880)
    at Array.map (<anonymous>)
    at load.L.load.f.then (es-module-shims.js:879)
```

**Comment by Hypercubed** [comment]
This also throws an error:

```
importShim('https://unpkg.com/lit-html@1.0.0/lit-html.js').then(x => console.log(x));
```

**Comment by bkardell** [comment]
Using the code from /dist right now this seems to work for me?

https://bramble-veil.glitch.me/other.html

**Comment by Hypercubed** [comment]
@bkardell I think your demo is not the same... importing a full non-relative URL (with https://) works via the import map... but not when the URL is used directly.

**Comment by bkardell** [comment]
Ah, yes, I see... This one captures it? https://bramble-veil.glitch.me/top.html

As I understand it, _most_ things may not be in the import map and the idea is that they would then 'fall through', but it looks like there is no falling through unaccosted in `resolveIfNotPlainOrUrl(..)`.

This file is using a slight tweak of the file, just as I worked through to understand how all this was working which adds a test for ones that look like a URL (https://glitch.com/edit/#!/bramble-veil?path=shimport/es-module-shims-temp.js:18:0), but that's just a naive test to check that theory real quick, I don't think it is right.  If I understand the proposal in wicg, you could technically remap a URL too (which could be useful for example in node ...maybe? idk), so I think it has more to do with whether there is an existing entry in the map or not.

Anyways, I'm keen on following this a little closer and potentially contributing as I wrap my head around it all - I think it looks like work on getting at least the mapping real solid is going on in the wicg repo?

**Comment by guybedford** [comment]
Published in 0.2.1.
